models/classes/runner/toolsStates/RdsToolsStateStorage.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2014 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoQtiTest\models\runner\toolsStates;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use Doctrine\DBAL\Query\QueryBuilder;
+use oat\taoResultServer\models\classes\ResultDeliveryExecutionDelete;
+use \core_kernel_classes_Resource;
+use oat\oatbox\service\ConfigurableService;
+
+class RdsToolsStateStorage extends ToolsStateStorage
+{
+    /**
+     * Constants for the database creation and data access
+     *
+     */
+    const TABLE_NAME = "runner_tool_states";
+    const DELIVERY_EXECUTION_ID_COLUMN = 'delivery_execution_id';
+    const TOOL_NAME_COLUMN = 'tool_name';
+    const TOOL_STATE_COLUMN = 'tool_state';
+
+    /**
+     * @return \common_persistence_SqlPersistence
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     */
+    protected function getPersistence()
+    {
+        $persistenceId = $this->hasOption(self::OPTION_PERSISTENCE) ?
```

Comment:
`hasOption()` does an `isset($this->options[$name])` check. This check is also performed in `getOption()`, so this can be put in a oneliner like: `$persistenceId = $this->getOption(self::OPTION_PERSISTENCE) ?: 'default';`